app/models/consolidated_health_answers.rb

```diff
@@ -0,0 +1,30 @@
+class ConsolidatedHealthAnswers
+  def initialize
+    @answers = {}
+  end
+
+  def add_answer(responder:, question:, answer:, notes: nil)
+    @answers[question] ||= []
+    @answers[question] << { responder:, answer:, notes: }
+  end
+
+  def to_h
+    consolidated_answers
+  end
+
+  private
+
+  def consolidated_answers
+    @answers.transform_values { |answers| consolidate_answers(answers) }
+  end
+
+  def consolidate_answers(answers)
+    if answers.length == 1
+      answers
+    elsif answers.uniq { |a| [a[:answer], a[:notes]] }.length == 1
+      [answers.first.merge(responder: "All")]
+    else
+      answers
```

Comment:
I think I agree with the golfing attempt here, but it might also make it more readable to use a helper method for `answers.uniq { |a| [a[:answer], a[:notes]] }` somehow.

Reply:
Agree with suggestion, incorporated in eb0be58f083225739a519d7bbdd3a4d67d09125d